src/levenberg_marquardt.jl

```diff
@@ -36,13 +36,13 @@ function levenberg_marquardt(df::OnceDifferentiable, initial_x::AbstractVector{T
     lambda = T(10), tau=T(Inf), lambda_increase::Real = 10.0, lambda_decrease::Real = 0.1,
     min_step_quality::Real = 1e-3, good_step_quality::Real = 0.75,
     show_trace::Bool = false, lower::Vector{T} = Array{T}(undef, 0), upper::Vector{T} = Array{T}(undef, 0), avv!::Union{Function,Nothing,Avv} = nothing
-    ) where T
+    ) where T<:Real
 
     # First evaluation
     value_jacobian!!(df, initial_x)
-    
+
     if isfinite(tau)
-        lambda = tau*maximum(jacobian(df)'*jacobian(df))
+        lambda = tau*maximum(abs.(jacobian(df)'*jacobian(df)))
```

Comment:
you can use `maximum(abs, x)` to do that with one less allocation (not that it matters of course)

Reply:
Good idea